packages/flutter/lib/src/widgets/scroll_position_with_single_context.dart

```diff
@@ -188,6 +194,14 @@ class ScrollPositionWithSingleContext extends ScrollPosition implements ScrollAc
       curve: curve,
       vsync: context.vsync,
     );
+    // If an invocation of [hold] is being interrupted by [animateTo], whereby
+    // the [ScrollableState] has not yet received the handle to the
+    // [HoldActivity], it will not be properly disposed of in [beginActivity].
+    // Therefore, we dispose of any active hold here.
+    if (_currentHold != null) {
```

Comment:
And wouldn't we also have a similar problem when there is an ongoing drag? (e.g. when an animation triggers based on the onScrollStartNotification before we had a chance to get the handle for the drag)